### DISCUSSION TITLE
[5.x]: Disable automatic button rendering

### DISCUSSION THREAD

**Discussion by mikesnoeren** [solspace/craft-freeform#1073] [Questions & Answers] [answered]
In previous versions, I was able to detect the button as a field type, to render my own custom buttons (see code below). 
However, since V5, the button is no longer seen as a field. and it automatically rendered (in a custom template)

Please give an option to disable the automatic rendering of the button.
Allow us to do this:

```twig
{% switch field.type %}
{% case 'submit' or 'save' %}
  <div class="flex justify-end">
    {% include '_inc/helpers/button.twig' with {
      element: 'button',
      type: 'submit',
      content: field.label
    } %}
</div>
```

**Comment by kjmartens** [comment, marked as answer]
Hi @mikesnoeren,

You are correct that this has changed in Freeform 5, but there is a way to disable the automatic button rendering and set it yourself. 🙂 

You can [see the options available](https://docs.solspace.com/craft/freeform/v5/templates/formatting/#submit-button-customization) in the documentation, but you might likely wish to go this route:

``` twig
{{ form.renderTag({
    returnUrl: "contact/success",
    disable: ["submitButtons"]
}) }}

    {# Your fields #}

    {% set page = form.pages.current %}

    <button {{ page.buttons.submitRenderProps({ 'data-optional': 'attributes here' }) }}>
        Submit
    </button>
    {% if form.pages.current.buttons.back %}
        <button {{ page.buttons.backRenderProps }}>
            Back
        </button>
    {% endif %}
    {% if form.pages.current.buttons.save %}
        <button {{ page.buttons.saveRenderProps }}>
            Save
        </button>
    {% endif %}

{{ form.renderClosingTag }}
```

https://docs.solspace.com/craft/freeform/v5/templates/formatting/#submit-button-customization

**Comment by samuel-clara** [comment]
This seems to not work for multi page forms. At page 2 no buttons are shown.

**Reply by kjmartens** [reply]
What does your full template code look like? You need to include the buttons and such inside each page conditional like it is here:
https://docs.solspace.com/craft/freeform/v5/templates/queries/form/#manual-multi-page-forms
